app.py

```diff
@@ -0,0 +1,123 @@
+        'fields[1]': 'record_id',
+        'fields[2]': 'eligibility_screening_complete',
+        'fields[3]': 'consent_form_complete',
+        'fields[4]': 'enrollment_questionnaire_complete',
+        'filterLogic': filter_logic,
+        'rawOrLabel': 'raw',
+        'rawOrLabelHeaders': 'raw',
+        'exportCheckboxLabel': 'false',
+        'exportSurveyFields': 'false',
+        'exportDataAccessGroups': 'false',
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+
+    if len(response.json()) == 0:
+        return None
+
+    assert len(response.json()) == 1, \
+        f"Error: Multiple records matching {filter_logic}"
+
+    return response.json()[0]
+
+def register_net_id(net_id: str) -> str:
+    """
+    Returns the REDCap record ID of the participant newly registered under the
+    given *net_id*.
+    """
+    # REDCap enforces that we must provide a non-empty record ID. Because we're
+    # using `forceAutoNumber` in the POST request, we do not need to provide a
+    # real record ID.
+    values = [{'netid': net_id, 'record_id': 'record ID cannot be blank'}]
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'record',
+        'format': 'json',
+        'type': 'flat',
+        'overwriteBehavior': 'normal',
+        'forceAutoNumber': 'true',
+        'data': json.dumps(values),
+        'returnContent': 'ids',
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+    return response.json()[0]
+
+
+def generate_survey_link(record_id: str) -> str:
+    """
+    Returns a generated survey link to the eligibility screening instrument for
+    the given *record_id*.
+    """
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'surveyLink',
+        'format': 'json',
+        'instrument': 'eligibility_screening',
+        'event': 'enrollment_arm_1',  # TODO subject to change
+        'record': record_id,
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+    return response.text
+
+
+@app.route('/')
+def main():
+    # Get NetID from Shibboleth data
+    net_id = request.remote_user
+
+    while not net_id:
+        # TODO: Redirect to Shibboleth login
+        net_id = 'KaasenG'
+        pass
+
+    user_data = fetch_user_data(net_id)
+
+    if user_data is None:
+        # If not in REDCap project, create new record
+        user_data = {'record_id': register_net_id(net_id)}
+
+    # TODO -- generate a survey link for a particular day
+    # We are awaiting finalization of the REDCap project to know how
+    # daily attestations (repeating instruments) will be implemented.
+    if is_complete('eligibility_screening', user_data) and \
+        is_complete('consent_form', user_data) and \
+        is_complete('enrollment_questionnaire', user_data):
```

Comment:
This can be shortened to:
```python
instruments_to_check = ['eligibility_screening', 'consent_form', 'enrollment_questionnaire']
if all(map(lambda x: is_complete(x, user_data), instruments_to_check))
```
Not sure which way reads easier 🤷

Reply:
I thought about using `all` but opted for verbosity. I don't have a strong preference either way.